src/main/java/horseshoe/TemplateRenderer.java

```diff
@@ -11,19 +12,35 @@
 	private final Template template;
 	private final String indentation;
 	private final Expression expression;
+	private final Function<Object, TemplateRenderer> fn;
 
-	TemplateRenderer(final Template template, final String indentation, final Expression expression) {
+	TemplateRenderer(final Template template, final String indentation, final Expression expression, final Function<Object, TemplateRenderer> fn) {
 		this.template = template;
 		this.indentation = indentation;
 		this.expression = expression;
+		this.fn = fn;
+	}
+
+	TemplateRenderer(final Template template, final String indentation, final Expression expression) {
+		this(template, indentation, expression, null);
 	}
 
 	TemplateRenderer(final Template template, final String indentation) {
-		this(template, indentation, null);
+		this(template, indentation, null, null);
+	}
+
+	TemplateRenderer(final Function<Object, TemplateRenderer> fn, final Expression expression) {
+		this(null, null, expression, fn);
 	}
 
 	@Override
 	public void render(final RenderContext context, final Writer writer) throws IOException {
+		if (fn != null) {
+			final TemplateRenderer renderer = fn.apply(expression.evaluate(context));
```

Comment:
If the expression results in `null`, can we just return at this point and not render the template? That seems to be more inline with the rest of Horseshoe than trying to lookup a template with the name "null".

Reply:
Done